### ISSUE TITLE
Support typeToString for evolving types

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#16230]
**TypeScript Version:**  nightly (2.4.0-dev.20170602)

**Code**

```ts
function f() {
    let x;
    x = 1;
    return x;
}
```

If I hover over one of the first two `x`s, I see `let x: any`. Most people are going to think that that means it's an implicit `any` that wasn't flagged by the compiler.

**Comment by DanielRosenwasser** [comment]
This is technically by design - in positions where `x`  is modified, it's treated as an `any`.

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
